File: packages/mobile/lib/domain/entities/borrow_entity.dart
import 'package:entrepoty/domain/domain.dart';

class BorrowEntity {
  int id;
  CustomerEntity customer;
  ProductEntity product;
  DateTime date;
  BorrowEntity({
    required this.id,
    required this.customer,
    required this.product,
    required this.date,
  });

  factory BorrowEntity.fromJson(Map json) => BorrowEntity(
        id: json['id'] ?? 0,
        customer: CustomerEntity.fromJson(json['customer'] ?? {}),
        product: ProductEntity.fromJson(json['product'] ?? {}),
        date: DateTime.parse(json['datetime']),
      );
}

File: packages/server/src/main/kotlin/com/entrepoty/Entrepoty/domain/entities/BorrowEntity.kt
package com.entrepoty.Entrepoty.domain.entities

import jakarta.persistence.*
import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction
import org.jetbrains.annotations.NotNull

@Entity
@Table(name = "borrow")
class BorrowEntity {

    @field:Id
    @field:GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long = 0

    @field:NotNull
    @field:ManyToOne
    @field:OnDelete(action = OnDeleteAction.CASCADE)
    lateinit var customer: Customer

    @field:NotNull
    @field:OneToOne
    @field:OnDelete(action = OnDeleteAction.CASCADE)
    lateinit var product: ProductDetailEntity

    @field:ManyToOne
    @field:OnDelete(action = OnDeleteAction.NO_ACTION)
    lateinit var createdBy: User

    @field:NotNull
    @field:Temporal(TemporalType.DATE)
    lateinit var datetime: java.sql.Date

    constructor(customer: Customer, product: ProductDetailEntity, datetime: java.sql.Date, createdBy: User){
        this.customer = customer
        this.product = product
        this.datetime = datetime
        this.createdBy = createdBy
    };
}